src/main/java/org/dynmap/towny/DynmapTownyPlugin.java

```diff
@@ -552,33 +553,64 @@ private void addStyle(Town town, String resid, String natid, AreaMarker m, TownB
         m.setRangeY(y, y);
         m.setBoostFlag(defstyle.getBoost(as, ns));
 
-        //If dynamic nation colors is enabled, read the color from the nation and town objects
-        try {
-            if(dynamicNationColorsEnabled) {
-                // Get the town colour for the line colour.
-                String townMapColorHexCode = town.getMapColorHexCode();
-                // Get the nation colour for the fill colour.
-                String nationMapColourHexCode = town.getNationMapColorHexCode();
-                
-                // Fall back to the town's colour for line colour if the nation returned null;
-                if (nationMapColourHexCode == null)
-                	nationMapColourHexCode = town.getMapColorHexCode();
-                
-                //Get colour as int
-                int townMapColorInteger = Integer.parseInt(townMapColorHexCode, 16);
-                int nationMapColorInteger = Integer.parseInt(nationMapColourHexCode, 16);
+        //Read dynamic colors from town/nation objects
+        if(dynamicTownColorsEnabled || dynamicNationColorsEnabled) {
+            try {
+                String colorHexCode; 
+                Integer townFillColorInteger = null;
+                Integer townBorderColorInteger = null;
 
-                //Set stroke style
-                double strokeOpacity = m.getLineOpacity();
-                int strokeWeight = m.getLineWeight();
-                m.setLineStyle(strokeWeight, strokeOpacity, nationMapColorInteger);
+                //CALCULATE FILL COLOUR
+                if (dynamicTownColorsEnabled) {
+                    //Here we know the server is using town colors. This takes top priority for fill
+                    colorHexCode = town.getMapColorHexCode();              
+                    if(!colorHexCode.isEmpty()) {
+                        //If town has a color, use it
+                        townFillColorInteger = Integer.parseInt(colorHexCode, 16);                
+                    }                
```

Comment:
The nation's map color could be used as a fallback here, in case the town has not set a map color yet

Reply:
- I considered that
- The only scenario this fallback could happen is when a server had already been using nation colours, and just upgraded the system to allow special town colours.
- In this scenario, players could be very familiar with the scheme of a single colour in each town, indicating nation allegiance.
- Assume as standard that the players & maybe staff do not read the release information on the town colour feature.
- With the indicated fallback, at first nothing would change, but they would soon see just ONE or TWO towns diverge from the familiar scheme, causing them to potentially think that a bug was happening, and causing scattered confusion.
- Without the indicated fallback, town fill instead falls back to the default dynmap colour.  Thus the server will immediately see ALL towns suddenly fill red, which will be an unmistakeable signal for them to wake up and notice/discuss the new feature, which I hope will be a better experience & introduction to the feature than the previous.